### PR TITLE
Enable validation on ConduitFacts object.

### PR DESCRIPTION
**Pull request by awood**

### CODE REVIEW

**Comment by mstead** [comment]
I'd like to have a quick chat about validation on the canonical facts that we intend on sending to inventory before we merge this.

File: src/main/java/org/candlepin/insights/inventory/ConduitFacts.java [hunk 1]
```diff
@@ -20,21 +20,39 @@
...
 /**
  * POJO that holds all facts scoped for collection by the conduit.
  */
 public class ConduitFacts {
     private UUID subscriptionManagerId;
     private UUID biosUuid;
-    private List<String> ipAddresses;
+
+    @NotEmpty
+    private List<@Length(min = 7, max = 45) String> ipAddresses;
```
Comment (mstead) [inline src/main/java/org/candlepin/insights/inventory/ConduitFacts.java]: Should we be enforcing @NotEmpty here? We should be only sending what we know.

File: src/main/java/org/candlepin/insights/inventory/ConduitFacts.java [hunk 2]
```diff
@@ -20,21 +20,39 @@
...
+
+    @Length(min = 1, max = 255)
     private String fqdn;
-    private List<String> macAddresses;
+
+    @NotEmpty
+    private List<@Length(min = 1, max = 23) String> macAddresses;
```
Comment (mstead) [inline src/main/java/org/candlepin/insights/inventory/ConduitFacts.java]: Should we be enforcing @NotEmpty here? We should be only sending what we know.

File: src/main/java/org/candlepin/insights/inventory/ConduitFacts.java (hunk 1 above)
Comment (awood) [reply src/main/java/org/candlepin/insights/inventory/ConduitFacts.java]: I'm working off of https://github.com/dehort/insights-host-inventory/blob/input_validation/swagger/api.spec.yaml#L372 which specifies a minimum length.

File: src/main/java/org/candlepin/insights/inventory/ConduitFacts.java (hunk 2 above)
Comment (awood) [reply src/main/java/org/candlepin/insights/inventory/ConduitFacts.java]: Similar to the previous comment: https://github.com/dehort/insights-host-inventory/blob/input_validation/swagger/api.spec.yaml#L394

File: src/main/java/org/candlepin/insights/inventory/ConduitFacts.java (hunk 1 above)
Comment (mstead) [reply src/main/java/org/candlepin/insights/inventory/ConduitFacts.java]: As discussed in scrum, we should at least remove the not null constraint.

File: src/main/java/org/candlepin/insights/inventory/ConduitFacts.java (hunk 2 above)
Comment (mstead) [reply src/main/java/org/candlepin/insights/inventory/ConduitFacts.java]: As discussed in scrum, we should at least remove the not null constraint.

**Comment by mstead** [comment]
ACK - Just need to make the gate happy. :/

**Pull request merged**